src/ui/common/Sidebar/SidebarComponent.tsx

```diff
@@ -0,0 +1,65 @@
+import Drawer from '@mui/material/Drawer';
+import List from '@mui/material/List';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemText from '@mui/material/ListItemText';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import HomeIcon from '@mui/icons-material/Home';
+import AdbIcon from '@mui/icons-material/Adb';
+import sidebarStyle  from '../../../styles/sidebarStyle';
+import { useNavigate } from 'react-router-dom';
+import { useTheme } from '@mui/material/styles';
+import { sideBarItems } from '../../../constants/sidebarItems';
+import { DrawerHeader } from './SidebarHeader';
+import { toggleStatus } from '../../../features/opener/sidebarSlice';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../../../app/store';
+
+/*This component uses Redux capabilities to access the storage (useSelector()), 
+as well as to change the state of the application (useDispatch()). 
+The useNavigate() hook is used to navigate the specified routes.*/
+
+const SidebarComponent = () => 
+{
+   const isOpened = useSelector((state: RootState) => state.sidebar.open)
+   const theme = useTheme();
+   const dispatch = useDispatch();
+   const navigate = useNavigate();
+   const handleSidebarOpen = () => 
+   {
```

Comment:
Incorrect formating.

Reply:
Application logic and state have been moved to containers to separate them from the component display. From the container to the component, state and logic are passed using props.